dynamic_instrumentation/src/instrument_memory.rs

```diff
@@ -237,12 +270,17 @@ impl<'a, 'tcx: 'a> Visitor<'tcx> for CollectFunctionInstrumentationPoints<'a, 't
                             None
                         };
 
+                        // Trace the base and field idx of each field projection
                         self.add_instrumentation_point(
                             location,
                             field_fn,
                             vec![
-                                Operand::Copy(base.local.into()),
-                                make_const(self.tcx, field.as_u32()),
+                                // XXX(fw): I don't understand what the type of `base` will be:
+                                // it has fields, so it should be a value rather than a pointer...
+                                // but it could be any type (e.g. a struct that cannot be cast to an
+                                // integer),so how can we always pass it to `ptr_field`?
+                                InstrumentationOperand::RawPtr(Operand::Copy(base.local.into())),
```

Comment:
That makes sense--because the `Operand` is `base.local` (not `base` itself), the type of the place represented by `base` (with projection) doesn't matter; only the type of `base.local` (which is, as you state, identical to `place.local`) matters.

I'll update the comment.